connectors/sources/confluence.py

```diff
@@ -929,10 +951,20 @@ async def search_by_query(self, query):
                 and entity_details.get("container", {}).get("title") is None
             ):
                 continue
+            ancestor_title = ""
+            if entity.get("entityType") != "space":
+                ancestors = await self.confluence_client.fetch_ancestors(
+                    page_id=entity_details.get("id")
+                )
+                ancestor_title = " ".join(
+                    ancestor.get("title", "")
+                    for ancestor in ancestors.get("ancestors", [])
+                )
```

Comment:
This is for advanced sync rules, if the user defined a specific query, right? In that case, I don't think we should execute extra requests. If they want the ancestors, they can include that in the query. When forming up the output document, we can just look to see if there's an `"ancestors"` key in the response.

Reply:
@seanstory But still if user right cql query, it will not fetch ancestors in sync rule because the API endpoint which we are using in sync rule has no query param to fetch ancestors directly. we need to do an additional API call for fetching it. What we can do - we can add a new boolean field in sync rules i.e. `is_ancestors` and if it's true we'll fetch ancestor otherwise not

Reply:
That surprised me, but I took a quick look, and couldn't find a way to get ancestors (in bulk) from the search API either.

In that case, I think we should just include in the docs and release notes that `"ancestors"` will not be present if you use advanced sync rules. I don't think the performance hit is worth it, and I don't really want to add the extra cost of another configuration to make sync rules or the RCFs more complex.